aws/data_source_aws_dx_connections.go

```diff
@@ -0,0 +1,65 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/directconnect"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/service/directconnect/finder"
+)
+
+func dataSourceAwsDxConnections() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsDxConnectionsRead,
+
+		Schema: map[string]*schema.Schema{
+			"ids": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func dataSourceAwsDxConnectionsRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).dxconn
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
+
+	tagsToMatch := keyvaluetags.New(d.Get("tags").(map[string]interface{})).IgnoreAws().IgnoreConfig(ignoreTagsConfig)
+
+	connections, err := finder.Connections(conn, &directconnect.DescribeConnectionsInput{})
+	if err != nil {
+		fmt.Errorf("error reading DirectConnect connections: %w", err)
```

Comment:
Fix `golangci` linter error:

```suggestion
		return fmt.Errorf("error reading DirectConnect connections: %w", err)
```